### PR TITLE
DP-149 - fix a problem with importing some QRs

### PR DESCRIPTION
**Pull request by yosriz**
#### Main purpose:
Fix an issue where some QRs cannot be imported and the user gets import error in restore flow.
#### Technical description:
* add TRY_HARDER, PURE_BARCODE flags to make sure we put max efforts decoding a QR...
* use QRCodeReader as it supposed to be faster
* add a test with a QR that cannot be decoded before
following this issue in zxing lib https://github.com/zxing/zxing/issues/995 
(thanks @steelbytes!)
#### Dilemmas you faced with? 
N/A
#### Tests:
add a test with a QR that cannot be decoded before
#### Documentation:
N/A

### CODE REVIEW

File: recovery/src/androidTest/java/com/kin/ecosystem/recovery/qr/QRBarcodeGeneratorImplTest.java [hunk 1]
```diff
@@ -73,10 +79,16 @@ public void generate_success() throws Exception {
 
 	@Test
 	public void decodeQR_success() throws Exception {
-		Bitmap bitmap = TestUtils.loadBitmapFromResource(this.getClass(), "qr_test.png");
+		// test a QR that was being able to be decoded only with TRY_HARDER flag on in zxing library
+		Bitmap bitmap = TestUtils.loadBitmapFromResource(this.getClass(), "backup_qr_try_harder.png");
```
Comment (YohayBar) [inline recovery/src/androidTest/java/com/kin/ecosystem/recovery/qr/QRBarcodeGeneratorImplTest.java:83]: backup_qr_try_harder.png is the QR that didn't succeed before?

**Review by YohayBar — APPROVED** [review]
look good to me - need to also QA test it

File: recovery/src/androidTest/java/com/kin/ecosystem/recovery/qr/QRBarcodeGeneratorImplTest.java (hunk 1 above)
Comment (yosriz) [reply recovery/src/androidTest/java/com/kin/ecosystem/recovery/qr/QRBarcodeGeneratorImplTest.java:83]: yep

**Comment by OrenZak** [comment]
@yosriz please sync with dev

**Pull request merged**